### ISSUE TITLE
Value of a PARAM_INT_OUT is not printed

### ISSUE THREAD

**Issue by alexandretea** [mlpack/mlpack#1100]
When an output parameter is defined using the macro PARAM_INT_OUT, the parameter is not printed even if a value has been set to it. I'm using the version 2.2.4 of mlpack.

The following code shows the problem:
```c++
#include <mlpack/core.hpp>

using namespace mlpack;

PARAM_DOUBLE_OUT("out_double", "desc");
PARAM_INT_OUT("out_int", "desc");

int main(int argc, char** argv)
{
    CLI::GetParam<double>("out_double") = 42.8484;
    CLI::GetParam<int>("out_int") = 13;
    return 0;
}
```

Running this code, only the double output parameter is printed.

**Comment by rcurtin** [comment]
Thanks for pointing this out.  I am about to release a bugfix 2.2.5, I'll include a fix for this also.

**Comment by rcurtin** [comment]
Thanks so much for the simple code to test the fix, it really is nice to get a simple program like that for debugging. :)

It turns out, the problem was quite simple: f77fe9524.  I released mlpack-2.2.5 with the fix: http://www.mlpack.org/files/mlpack-2.2.5.tar.gz.

Please feel free to report any issues in the future, and we'll be happy to try and fix them. :)
